src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/LanguageServices/CSharpLanguageFeaturesProvider.cs

```diff
@@ -122,13 +122,13 @@ public string ConcatNamespaces(params string[] namespaceNames)
 
         private static bool IsValidIdentifierChar(char c)
         {
-            var category = CharUnicodeInfo.GetUnicodeCategory(c);
+            UnicodeCategory category = CharUnicodeInfo.GetUnicodeCategory(c);
             return s_identifierCharCategories.Contains(category);
         }
 
         private static bool IsValidFirstIdentifierChar(char c)
         {
-            var category = CharUnicodeInfo.GetUnicodeCategory(c);
+            UnicodeCategory category = CharUnicodeInfo.GetUnicodeCategory(c);
```

Comment:
Ugh.  Seems apparent to me.  @kuhlenh I wonder if there are some more tweaks to the "apparent" heuristic:

1. Static named: "GetTypeName/CreateTypeName/etc" that return "TypeName"
2. See the "CreateBuilder" example below too for another example of this with nested types.

Reply:
This wouldn't be apparent; what relationship does UnicodeCategory have with CharUnicodeInfo?